src/edit/DistortableImage.Edit.js

```diff
@@ -319,40 +361,116 @@ L.DistortableImage.Edit = L.Handler.extend({
 	},
 
 	_hideMarkers: function() {
-		var currentHandle =
-      this._mode === "distort" ? this._distortHandles : this._rotateHandles;
+		var currentHandle = this._handles[this._mode];
 		currentHandle.eachLayer(function (layer) {
 			var drag = layer.dragging,
 				opts = layer.options;
 
 			layer.setOpacity(0);
 			if (drag) { drag.disable(); }
 			if (opts.draggable) { opts.draggable = false; }	
-		});
-		
+		});	
 	},
 	
 	// TODO: toolbar for multiple image selection
 	_showToolbar: function() {
 		var overlay = this._overlay,
-      // target = event.target,
-			map = overlay._map;
+			c1Idx = 0,
+			c2Idx = 1,
+			map = overlay._map,
+			ANCHOR = 25;
 
 		/* Ensure that there is only ever one toolbar attached to each image. */
 		this._hideToolbar();
+		overlay._reset();
+		var topLeft = '';
+		var topRight = '';
+		var insideOut = false;
+		if (overlay._isInsideOut()) { 
+			insideOut = true;
+			 topLeft = map.project(overlay.getCorners()[c2Idx], 13);
+			 topRight = map.project(overlay.getCorners()[c1Idx], 13);
+		} else {
+		// convert the corners to the correct projection (as Points) with zoom level
+				// insideOut = false;
+			  topLeft = map.project(overlay.getCorners()[c1Idx], 13);
+				topRight = map.project(overlay.getCorners()[c2Idx], 13);
+		}
 		
-		var point;
-		// if (event.containerPoint) { point = event.containerPoint; }
-		// point = target._leaflet_pos;
-		point = overlay._image._leaflet_pos;
-		
-		var raised_point = map.containerPointToLatLng(new L.Point(point.x,point.y-20));
-		raised_point.lng = overlay.getCenter().lng;
+		// place toolbar at their center and anchor it
+		var toolPoint = overlay._calcCenterTwoCornerPoints(topLeft, topRight);
+		toolPoint.y -= ANCHOR;
+
+		// window.topRight = topRight;
+		// window.topLeft = topLeft;
 	
+		var dx = Math.abs(topRight.x - topLeft.x),
+			dy = topRight.y - topLeft.y;
+
+		var angle = -(L.TrigUtil.calcAngleDegrees(dx, dy));
+
+		window.angle = angle;
+
+		var raised_point = map.unproject(toolPoint, 13);
+
 		if (this._overlay.options.suppressToolbar !== true) {
 			this.toolbar = new L.DistortableImage.EditToolbar(raised_point).addTo(map, overlay);
 			overlay.fire('toolbar:created');
 		}
+
+		// if (this._mode !== 'rotate') {
+		
+		
+
+		if ((overlay._isSidewaysRight() && !(overlay._isRightSideUp())) && (!(overlay._isUpsideDown()) && topLeft.x < topRight.x)) {
+			console.log("here1");
+			if (!insideOut) {
+				angle = -(angle + 180);
+			} else {
+				angle = -angle + 180;
+				angle = angle + 90;
+				angle -= 90;
+			}
+			this.toolbar._rotateToolbarAngleDeg(angle);
+			return; 
+		}
+
+		if (overlay._isUpsideDown()) {
+			console.log("herelog");
+			angle = -(angle) + 180;
+			this.toolbar._rotateToolbarAngleDeg(angle);
+			return; 
+		}
+
+		if ((overlay._isSidewaysLeft()) && (!(overlay._isUpsideDown()) && (topRight.x > topLeft.x))) {
+			console.log("hereleft");
+			if (!insideOut) {
+				console.log("notinsideout");
+					angle = -angle + 180;
```

Comment:
Did you mean to write this here (else the `if-else` blocks will give the same output)?
```suggestion
					angle = -(angle + 180);
```